mule-deployer/src/main/java/org/mule/tools/client/AbstractMuleClient.java

```diff
@@ -284,9 +294,15 @@ public String getBusinessGroupIdByBusinessGroupPath() {
     return currentOrgId;
   }
 
-  private String getBearerToken(Credentials credentials) {
+  private String getBearerToken(AnypointCredential credentials) {
     if (isBlank(bearerToken)) {
-      bearerToken = authenticationServiceClient.getBearerToken(credentials);
+      switch (credentials.credentialType()) {
+        case user:
+          Credentials creds = (Credentials) credentials;
+          bearerToken = authenticationServiceClient.getBearerToken(creds);
+        case token:
+          bearerToken = ((AnypointToken) credentials).getToken();
```

Comment:
```suggestion
          bearerToken = ((AnypointToken) credentials).getToken();
          break;
```

Reply:
Thanks! I have added next commit with this fix.